src/app/static/src/app/store/colourScales/actions.ts

```diff
@@ -0,0 +1,15 @@
+import {ActionContext, ActionTree} from "vuex";
+import {RootState} from "../../root";
+import {ColourScaleSelections, ColourScalesState} from "./colourScales";
+
+export interface ColourScalesActions {
+    updateSelectedSAPColourScales: (store: ActionContext<ColourScalesState, RootState>, colourScales: ColourScaleSelections) => void
+}
+
+
+export const actions: ActionTree<ColourScalesState, RootState> & ColourScalesActions = {
+    updateSelectedSAPColourScales(context, colourScales) {
+        const {commit, rootState} = context;
```

Comment:
Since there's nothing async going on here and it just calls straight through to the commit, can we just dispense with this action altogether and directly call the commit from the component using the `mapCommit`helper?

Reply:
I see about the rootState - but this is called from a component that already know what the selected data type is - could parameterise the mutation with `selectedDataType` as well as `colourScales`?

Reply:
(also yeah, mutations clearly can't access rootState so no idea what we were both remembering!)